### ISSUE TITLE
Improved logging to improve ambiguous error messages

### ISSUE THREAD

**Issue by tylerpotts** [jupyterhub/jupyterhub#3805]
### Proposed change

I'm using docker-compose to deploy my jupyterhub setup. In the process of upgrading one of my docker images, I ran into this error:

```
jupyterhub-1      | [D 2022-02-22 23:32:03.885 JupyterHub dockerspawner:982] Getting container 'jupyter-admin'
jupyterhub-1      | [I 2022-02-22 23:32:03.890 JupyterHub dockerspawner:988] Container 'jupyter-admin' is gone
jupyterhub-1      | [E 2022-02-22 23:32:03.896 JupyterHub user:800] Unhandled error starting admin's server: unsupported operand type(s) for +: 'NoneType' and 'list
```
When looking into the line number in `user:800` [here](https://github.com/jupyterhub/jupyterhub/blob/main/jupyterhub/user.py#L799) the logger doesn't give a full traceback of the error. This line only allowed me to see where the error was raised, not the line of code that raised it. Increasing the log level of jupyterhub didn't improve the situation.

I ended up needing to modify the source code and importing the `traceback` module in order to diagnose the problem. Here was my solution:

``` 
            else:
                self.log.error(
                    "Unhandled error starting {user}'s server: {error}".format(
                        user=self.name, error=e
                    )
                )
                self.log.error(traceback.format_exc()) # my code here
```

After a somewhat painful process of getting this modified version of jupyterhub into my image, I was able to see this more useful error message:

```
jupyterhub-1      | [E 2022-02-23 00:01:54.284 JupyterHub user:800] Unhandled error starting admin's server: unsupported operand type(s) for +: 'NoneType' and 'list'
jupyterhub-1      | [E 2022-02-23 00:01:54.284 JupyterHub user:805] Traceback (most recent call last):
jupyterhub-1      |       File "/opt/conda/envs/jupyterhub/lib/python3.9/site-packages/jupyterhub/user.py", line 717, in spawn
jupyterhub-1      |         url = await gen.with_timeout(timedelta(seconds=spawner.start_timeout), f)
jupyterhub-1      |       File "/opt/conda/envs/jupyterhub/lib/python3.9/site-packages/dockerspawner/dockerspawner.py", line 1270, in start
jupyterhub-1      |         obj = await self.create_object()
jupyterhub-1      |       File "/opt/conda/envs/jupyterhub/lib/python3.9/site-packages/dockerspawner/dockerspawner.py", line 1115, in create_object
jupyterhub-1      |         command=(await self.get_command()),
jupyterhub-1      |       File "/opt/conda/envs/jupyterhub/lib/python3.9/site-packages/dockerspawner/dockerspawner.py", line 1015, in get_command
jupyterhub-1      |         return cmd + self.get_args()
jupyterhub-1      |     TypeError: unsupported operand type(s) for +: 'NoneType' and 'list'
```

This helped me realize that my new docker image somehow had the `CMD` set to `null`, instead of an empty list `[]` :facepalm: 

### Who would use this feature?

All users

### (Optional): Suggest a solution

There may be a better solution than traceback to handle this, I'm not sure. It would be nice if the logger provided the ability to see the full traceback of errors so that modifying the source code isn't necessary.

**Comment by yuvipanda** [comment]
This was fixed in https://github.com/jupyterhub/jupyterhub/pull/3882 (which I somehow have no memory of lol).

**Comment by Strice91** [comment]
@tylerpotts I experience the same error currently. 

How did you solve the problem that your new docker images had the `CMD` set incorrectly?
I added this at the end of my local Dockerfile but it did not help:

```
...
CMD []
```

When I set the command in the **jupyter_config.py**, then, the error goes away, but the jupyterlab does not start anymore. I tried
- `c.DockerSpawner.cmd = []`
- `c.DockerSpawner.cmd = ["tini" "-g" "--" "start.sh"]`
